circuit/types/scalar/src/helpers/to_bits.rs

```diff
@@ -18,29 +18,27 @@ impl<E: Environment> ToBits for Scalar<E> {
     type Boolean = Boolean<E>;
 
     /// Outputs the little-endian bit representation of `self` *without* trailing zeros.
-    fn to_bits_le(&self) -> Vec<Self::Boolean> {
-        (&self).to_bits_le()
+    fn write_bits_le(&self, vec: &mut Vec<Self::Boolean>) {
+        (&self).write_bits_le(vec);
     }
 
     /// Outputs the big-endian bit representation of `self` *without* leading zeros.
-    fn to_bits_be(&self) -> Vec<Self::Boolean> {
-        (&self).to_bits_be()
+    fn write_bits_be(&self, vec: &mut Vec<Self::Boolean>) {
+        (&self).write_bits_be(vec);
     }
 }
 
 impl<E: Environment> ToBits for &Scalar<E> {
     type Boolean = Boolean<E>;
 
     /// Outputs the little-endian bit representation of `self` *without* trailing zeros.
-    fn to_bits_le(&self) -> Vec<Self::Boolean> {
-        self.bits_le
+    fn write_bits_le(&self, vec: &mut Vec<Self::Boolean>) {
+        vec.extend_from_slice(self.bits_le
             .get_or_init(|| {
                 // Note: We are reconstituting the scalar field into a base field.
                 // This is safe as the scalar field modulus is less than the base field modulus,
                 // and thus will always fit within a single base field element.
-                debug_assert!(
-                    console::Scalar::<E::Network>::size_in_bits() < console::Field::<E::Network>::size_in_bits()
-                );
+                debug_assert!(console::Scalar::<E::Network>::size_in_bits() < console::Field::<E::Network>::size_in_bits());
```

Comment:
(there are no changes here, it's just a bit `rustfmt` squeezed that I missed in my new "un-formatting" commit)